solutionbox/code_free_ml/mltoolbox/code_free_ml/trainer/task.py

```diff
@@ -58,41 +58,120 @@
 PG_CLASSIFICATION_SCORE_TEMPLATE = 'score_%s'
 
 
+class DatalabParser():
+  """Datalab parser.
+
+  When using Datalab magic's to run this trainer, it prints it's own help menu
+  that describes the required options that are common to all trainers. In order
+  to print just the options that are unique to this trainer, datalab calls this
+  file with --lite-help.
+
+  This class implements --lite-help by building a 2nd argparser that only has
+  the unique parameters. This insures the formatting is consistant with
+  argparse as argparse is used.
+  """
+
+  def __init__(self, epilog=None):
+    # full_parser is used by this trainer. lite_parser is only used to print
+    # a shorter help string when --lite-help is used.
+    self.full_parser = argparse.ArgumentParser(epilog=epilog)
+    self.lite_parser = argparse.ArgumentParser(usage='', epilog=epilog, add_help=False)
+
+    # The arguments added here are required to exist by Datalab's magics.
+
+    # Datalab help string
+    self.full_parser.add_argument(
+        '--lite-help', action=self.make_lite_help_action(),
+        help='Show a smaller help message and exit')
+
+    # I/O file parameters
+    self.full_parser.add_argument(
+        '--train-data-paths', type=str, required=True, action='append')
+    self.full_parser.add_argument(
+        '--eval-data-paths', type=str, required=True, action='append')
+    self.full_parser.add_argument('--job-dir', type=str, required=True)
+    self.full_parser.add_argument(
+        '--analysis-output-dir', type=str, required=True,
+        help=('Output folder of analysis. Should contain the schema, stats, and '
+              'vocab files. Path must be on GCS if running cloud training.'))
+    self.full_parser.add_argument(
+        '--run-transforms', action='store_true', default=False,
+        help='If used, input data is raw csv that needs transformation.')
+
+    # HP parameters
+    self.full_parser.add_argument(
+        '--learning-rate', type=float, default=0.01, help='optimizer learning rate')
```

Comment:
I think learning-rate, max-steps, num-epochs, train-batch-size, eval-batch-size, min-eval-frequency, save-checkpoints-secs also belongs to lite help. They may not apply in another package. Also, in Datalab, the only parameters that will always be passed to command line would be input data, output dir and run-transforms flag. These are required parameters. Everything else is passed through to the package itself.